src/python/aqueduct_executor/operators/connectors/data/s3.py

```diff
@@ -38,14 +38,15 @@ def authenticate(self) -> None:
                 % str(e)
             )
 
-        # Check that any user-supplied root directory exists.
+        # Check that any user-supplied root directory exists. If the object key name does not exist,
+        # create a new one.
         if self.root_dir != "":
             # If nothing is returned by this filter call, then the directory does not exist.
             if len(list(self.s3.Bucket(self.bucket).objects.filter(Prefix=self.root_dir))) == 0:
-                raise Exception(
-                    "Supplied root directory `%s` does not exist in bucket %s."
-                    % (self.root_dir, self.bucket)
-                )
+                if not self.root_dir.endswith("/"):
+                    self.root_dir = self.root_dir + "/"
+                bucket = self.s3.Bucket(self.bucket)
+                bucket.put_object(Bucket=self.bucket, Key=self.root_dir)
```

Comment:
Curious what happens if the user doesn't sufficient permissions? I'd like us to surface that error a little more nicely here.